src/mscorlib/src/System/Buffer.cs

```diff
@@ -256,325 +256,162 @@ internal unsafe static void Memcpy(byte* dest, byte* src, int len)
+            *(int*)(dest + 12) = *(int*)(src + 12);             // [0,16]
 #endif
-                    *(short*)(dest + 16) = *(short*)(src + 16);
-                    return;
-                case 19:
+            if (len <= 32) goto MCPY01;
 #if BIT64
-                    *(long*)dest = *(long*)src;
-                    *(long*)(dest + 8) = *(long*)(src + 8);
+            *(Block16*)(dest + 16) = *(Block16*)(src + 16);     // [0,32]
 #else
-                *(int*)dest = *(int*)src;
-                *(int*)(dest + 4) = *(int*)(src + 4);
-                *(int*)(dest + 8) = *(int*)(src + 8);
-                *(int*)(dest + 12) = *(int*)(src + 12);
+            *(int*)(dest + 16) = *(int*)(src + 16);
+            *(int*)(dest + 20) = *(int*)(src + 20);
+            *(int*)(dest + 24) = *(int*)(src + 24);
+            *(int*)(dest + 28) = *(int*)(src + 28);             // [0,32]
 #endif
-                    *(short*)(dest + 16) = *(short*)(src + 16);
-                    *(dest + 18) = *(src + 18);
-                    return;
-                case 20:
+            if (len <= 48) goto MCPY01;
 #if BIT64
-                    *(long*)dest = *(long*)src;
-                    *(long*)(dest + 8) = *(long*)(src + 8);
+            *(Block16*)(dest + 32) = *(Block16*)(src + 32);     // [0,48]
 #else
-                *(int*)dest = *(int*)src;
-                *(int*)(dest + 4) = *(int*)(src + 4);
-                *(int*)(dest + 8) = *(int*)(src + 8);
-                *(int*)(dest + 12) = *(int*)(src + 12);
+            *(int*)(dest + 32) = *(int*)(src + 32);
+            *(int*)(dest + 36) = *(int*)(src + 36);
+            *(int*)(dest + 40) = *(int*)(src + 40);
+            *(int*)(dest + 44) = *(int*)(src + 44);             // [0,48]
 #endif
-                    *(int*)(dest + 16) = *(int*)(src + 16);
-                    return;
-                case 21:
+            
+            MCPY01:
 #if BIT64
-                    *(long*)dest = *(long*)src;
-                    *(long*)(dest + 8) = *(long*)(src + 8);
+            *(Block16*)(destEnd - 16) = *(Block16*)(srcEnd - 16);
 #else
-                *(int*)dest = *(int*)src;
-                *(int*)(dest + 4) = *(int*)(src + 4);
-                *(int*)(dest + 8) = *(int*)(src + 8);
-                *(int*)(dest + 12) = *(int*)(src + 12);
+            *(int*)(destEnd - 16) = *(int*)(srcEnd - 16); 
+            *(int*)(destEnd - 12) = *(int*)(srcEnd - 12); 
+            *(int*)(destEnd - 8) = *(int*)(srcEnd - 8);
+            *(int*)(destEnd - 4) = *(int*)(srcEnd - 4);
 #endif
-                    *(int*)(dest + 16) = *(int*)(src + 16);
-                    *(dest + 20) = *(src + 20);
-                    return;
-                case 22:
-#if BIT64
-                    *(long*)dest = *(long*)src;
-                    *(long*)(dest + 8) = *(long*)(src + 8);
+            return;
+            
+            MCPY02:
+            if ((len & 24) == 0) goto MCPY03;
```

Comment:
Interesting that replacing the switch table with all of these branches results in better perf.